zerver/models.py

```diff
@@ -155,7 +156,27 @@ def as_sql(
         return f"{lhs} & {rhs} != 0", lhs_params + rhs_params
 
 
-def query_for_ids(query: QuerySet, user_ids: List[int], field: str) -> QuerySet:
+ModelT = TypeVar("ModelT", bound=models.Model)
+RowT = TypeVar("RowT")
+
+
+@overload
+def query_for_ids(query: QuerySet[ModelT], user_ids: List[int], field: str) -> QuerySet[ModelT]:
```

Comment:
Since django-stubs defines `QuerySet[ModelT]` to be the same thing as `ValuesQuerySet[ModelT, ModelT]`, this overload should be unnecessary.

Reply:
Fixed.